airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/stream_reader.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import datetime
+import logging
+import stat
+from io import IOBase
+from typing import Iterable, List, Optional
+
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from source_sftp_bulk.client import SFTPClient
+from source_sftp_bulk.spec import SourceSFTPBulkSpec
+
+
+class SourceSFTPBulkStreamReader(AbstractFileBasedStreamReader):
+    def __init__(self):
+        super().__init__()
+        self._sftp_client = None
+
+    @property
+    def config(self) -> SourceSFTPBulkSpec:
+        return self._config
+
+    @config.setter
+    def config(self, value: SourceSFTPBulkSpec):
+        """
+        FileBasedSource reads the config from disk and parses it, and once parsed, the source sets the config on its StreamReader.
+
+        Note: FileBasedSource only requires the keys defined in the abstract config, whereas concrete implementations of StreamReader
+        will require keys that (for example) allow it to authenticate with the 3rd party.
+
+        Therefore, concrete implementations of AbstractFileBasedStreamReader's config setter should assert that `value` is of the correct
+        config type for that type of StreamReader.
+        """
+        assert isinstance(value, SourceSFTPBulkSpec)
+        self._config = value
+
+    @property
+    def sftp_client(self) -> SFTPClient:
+        if self._sftp_client is None:
+            authentication = (
+                {"password": self.config.credentials.password}
+                if self.config.credentials.auth_type == "password"
+                else {"private_key": self.config.credentials.private_key}
+            )
+            self._sftp_client = SFTPClient(
+                host=self.config.host,
+                username=self.config.username,
+                **authentication,
+                port=self.config.port,
+            )
+        return self._sftp_client
+
+    def get_matching_files(
+        self,
+        globs: List[str],
+        prefix: Optional[str],
+        logger: logging.Logger,
+    ) -> Iterable[RemoteFile]:
+        # TODO: check prefixes
```

Comment:
Should these lines be deleted or uncommented?

Reply:
deleted